tests/nfs_hook_test.go

```diff
@@ -0,0 +1,408 @@
+				_, err = Client.getPVC(openebsNamespace, "nfs-"+pvcObj.Spec.VolumeName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isBackendPVCExist = false
+					break
+				}
+				Expect(err).To(BeNil(), "while fetching backend pvc")
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isBackendPVCExist).To(BeFalse(), "backend pvc should not exist after deleting nfs pvc")
+
+			By("verify deletion of NFS PV")
+			isNFSPVExist := true
+			for retries = 0; retries < maxRetryCount; retries++ {
+				_, err = Client.getPV(pvcObj.Spec.VolumeName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isNFSPVExist = false
+					break
+				}
+				Expect(err).To(BeNil(), "while fetching NFS PV")
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isNFSPVExist).To(BeFalse(), "NFS PV should not exist after deleting nfs pvc")
+		})
+	})
+
+	When("Hook configMap removed from provisioner deployment", func() {
+		It("should update the provisioner deployment", func() {
+			By("fetching provisioner deployment")
+			deploy, err := Client.getDeployment(OpenEBSNamespace, NFSProvisionerName)
+			Expect(err).To(BeNil(), "while fetching deployment %s/%s", openebsNamespace, NFSProvisionerName)
+
+			By("updating the provisioner deployment")
+			idx := 0
+			for idx < len(deploy.Spec.Template.Spec.Containers[0].Env) {
+				if deploy.Spec.Template.Spec.Containers[0].Env[idx].Name == string(provisioner.NFSHookConfigMapName) {
+					break
+				}
+				idx++
+			}
+			deploy.Spec.Template.Spec.Containers[0].Env = append(deploy.Spec.Template.Spec.Containers[0].Env[:idx], deploy.Spec.Template.Spec.Containers[0].Env[idx+1:]...)
+			_, err = Client.updateDeployment(deploy)
+			Expect(err).To(BeNil(), "while updateingupdating deployment %s/%s", openebsNamespace, NFSProvisionerName)
+
+			By("waiting for deployment rollout")
+			err = Client.waitForDeploymentRollout(OpenEBSNamespace, NFSProvisionerName)
+			Expect(err).To(BeNil(), "while verifying deployment rollout")
+		})
+	})
+
+	When("nfs hook configmap is deleted", func() {
+		It("should delete a configmap ", func() {
+			err := Client.deleteConfigMap(openebsNamespace, hookConfigMapName)
+			Expect(err).To(BeNil(), "while deleting hook configmap")
+		})
+	})
+})
+
+func annotationExist(objMeta *metav1.ObjectMeta, annotations map[string]string) bool {
+	annExist := true
+
+	for k, v := range annotations {
+		if objMeta.Annotations == nil {
+			annExist = false
+			break
+		}
+
+		ev, ok := objMeta.Annotations[k]
+		if !ok {
+			annExist = false
+			break
+		}
+
+		if ev != v {
+			annExist = false
+			break
+		}
+	}
+
+	return annExist
+}
+
+func finalizerExist(objMeta *metav1.ObjectMeta, finalizers []string) bool {
+	for _, finalizer := range finalizers {
+		finExist := false
+		for _, existingFinalizer := range objMeta.Finalizers {
+			if existingFinalizer == finalizer {
+				finExist = true
+			}
+		}
+
+		if finExist == false {
+			return false
+		}
+	}
+
+	return true
```

Comment:
nit/optional: Instead of looping one over the other we can do in following way:
```go
if len(finalizers) != len(objectMeta.Finalizers) {
   return false
}

desiredFinalizersMap := map[string]struct{}{}
for _, f := range finalizers {
   desiredFinalizersMap[f] = struct{}
}
for _, f := range objectMeta.Finalizers {
   if _, isFinalizerExist := desiredFinalizersMap[f]; !isFinalizerExist {
       return false
   }
  return true
}
```

Reply:
updated